HandEyeCalibration/Transform/Transform.cpp

```diff
@@ -0,0 +1,330 @@
+/*
+Example to show transformations to/from Transformation Matrix
+
+Zivid primarily operate with a (4x4) Transformation Matrix (Rotation Matrix + Translation Vector). 
+This example shows how to use Eigen to convert to and from:
+  AxisAngle, Rotation Vector, Roll-Pitch-Yaw, Quaternion
+
+ It provides convenience functions that can be reused in applicable applications.
+*/
+
+#include <Eigen/Core>
+#include <Eigen/Dense>
+#include <Eigen/Geometry>
+
+#include <opencv2/core/core.hpp>
+
+#include <iostream>
+
+enum class RotationConvention
+{
+    ZYX_Intrinsic,
+    XYZ_Extrinsic,
+    XYZ_Intrinsic,
+    ZYX_Extrinsic,
+    NOF_ROT
+};
+constexpr size_t nofRotationConventions = static_cast<size_t>(RotationConvention::NOF_ROT);
+
+struct RollPitchYaw
+{
+    RotationConvention convention;
+    Eigen::Array3d rollPitchYaw;
+};
+
+struct Representations
+{
+    Eigen::Vector4d angleAxis;
+    Eigen::Vector3d rotationVector;
+    Eigen::Vector4d quaternion;
+    std::array<RollPitchYaw, nofRotationConventions> rotations;
+};
+
+cv::Mat eigenToCv(const Eigen::MatrixXd &);
+Eigen::MatrixXd cvToEigen(const cv::Mat &);
+Eigen::Vector4d reorderAxisAngle(const Eigen::AngleAxisd &axisAngle);
+Eigen::Array3d rotmat2rpy(const Eigen::Matrix3d &rotationMatrix, const RotationConvention &rotation);
+Eigen::Matrix3d rpy2rotmat(const Eigen::Array3d &rollPitchYaw, const RotationConvention &rotation);
+std::string toString(RotationConvention convention);
+void compareRotation(const Eigen::Matrix3d &rotationMatrix1, const Eigen::Matrix3d &rotationMatrix2);
+void compareRotationExtended(const Eigen::Array3d &rollPitchYaw,
+                             const Eigen::Matrix3d &rotationMatrix,
+                             const RotationConvention &rotation);
+Representations transformFromZividToRobot(const Eigen::Matrix3d &);
+void transformFromRobotToZivid(const Eigen::Matrix3d &, const Representations &);
```

Comment:
I think the only reason there's a return value from the first one is because it's needed as input for the second one. Not sure we need a return value from the second one if we're not using it for anything?

Reply:
If we separate out writing results to file, then it makes sense to return the results.

This would however mean that I would have to add Eigen::Matrix3d for rotation matrix to the RollPitchYaw struct. This to best maintain relationship between convention and conversion. I'm not sure if that's going to be nice... As it is now `Representations` contain the results various conversion from Rotation Matrix ("Zivid format"). Adding Rotation Matrix to this one would change this.. and I'm not a fan.

I recall discussing this with myself, and I did actually consider separating out saving to file...

Maybe a compromise is to return an array of a new struct that combines rotation convention with rotation matrix (to maintain the relationship between euler angles and rotation matrix)?

Reply:
I'm not convinced we need a return value at all. What do you think @runenordmo?

Reply:
How about extracting write to file from `transformFromRobotToZivid`, but still call it from `transformFromRobotToZivid`?